sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAPITest.groovy

```diff
@@ -3364,6 +3367,53 @@ class FileAPITest extends APISpec {
         null     | null       | null        | null         | garbageLeaseID
     }
 
+    @Unroll
+    def "Schedule deletion"() {
+        given:
+        def fileClient = fsc.getFileClient(generatePathName())
+        fileClient.create()
+
+        when:
+        fileClient.scheduleDeletionWithResponse(fileScheduleDeletionOptions, null, null)
+        def expiryTimeProperty = fileClient.getProperties().getExpiresOn()
+
+        then:
+        (expiryTimeProperty != null) == hasExpiry
+
+        where:
+        fileScheduleDeletionOptions                                                             | hasExpiry
+        new FileScheduleDeletionOptions(Duration.ofDays(1), FileExpirationOffset.CREATION_TIME) | true
+        new FileScheduleDeletionOptions(Duration.ofDays(1), FileExpirationOffset.NOW)           | true
+        new FileScheduleDeletionOptions()                                                       | false
+        null                                                                                    | false
+    }
+
+    def "Schedule deletion time"() {
+        given:
+        def fileScheduleDeletionOptions = new FileScheduleDeletionOptions(getUTCNow().plusDays(1))
+        def fileClient = fsc.getFileClient(generatePathName())
+        fileClient.create()
+
+        when:
+        fileClient.scheduleDeletionWithResponse(fileScheduleDeletionOptions, null, null)
+        def expiryTimeProperty = fileClient.getProperties().getExpiresOn()
+
+        then:
+        expiryTimeProperty
```

Comment:
Can we verify the value of the time?

Reply:
yeah I can do something like check that time is within 5 min of current time or something

Reply:
Done